docs/operations.rst

```diff
@@ -0,0 +1,350 @@
+
+Tahoe-LAFS Storage Service
+--------------------------
+
+A Tahoe-LAFS storage service runs alongside the introducer.
+This mediates access to the customer's S3 bucket.
+
+Bottom-Up Components
+~~~~~~~~~~~~~~~~~~~~
+
+Looking at the service from the bottom up, the components involved in operating it are as folllows.
+
+AWS EC2
+-------
+
+The service runs on several EC2 instances.
+The EC2 instances used to run infrastructure are disposable.
+So long as enough of them are running to handle the service load, the service should operate properly.
+The EC2 instances run within a LeastAuthority-owned AWS account.
+
+Persistent state related to the operation of the service
+(such as customer subscriptions, server logs, monitoring data - **not** customer-owned, Tahoe-LAFS-managed data)
+is stored on EBS instances attached to the EC2 instances.
+The EBS instances belong to a LeastAuthority-owned AWS account.
+
+Each customer has their own EC2 instance.
+Customer data is stored on S3, not instance storage of EBS.
+Recreating a destroyed customer EC2 instance is possible, though it is a manual task.
+
+AWS EC2 is not a very tightly coupled component of the system.
+The signup process does integrate against the EC2 APIs directly.
+A future direction is for customers to receive service from a container instead of an EC2 instance.
+This will reduce coupling.
+
+AWS S3
+------
+
+Tahoe-LAFS services are configured with the S3 storage backend and therefore use S3 buckets.
+All customer-owned data resides in these S3 buckets.
+The S3 buckets belong to a LeastAuthority-owned AWS account.
+
+AWS S3 is a fairly tightly coupled component of the system.
+The Tahoe-LAFS S3 storage backend us leveraged extensively.
+Customer data is stored in S3 buckets.
+A transition to another storage system requires both code changes and data migration.
+
+Docker Registry
+---------------
+
+The service launches Docker containers using images hosted on a local Docker registry.
+
+It is possible for the images to be signed and the signatures to be checked before the containers are started.
+This is not currently implemented.
+
+It is possible for images to be stored on an encrypted EBS volume.
+This is not currently implemented.
+
+Kubernetes
+----------
+
+Some errors are expected as kubernetes resources are created.
+Ordering of creation of different components is not enforced.
+The cluster should converge on a working state as dependencies get created.
+
+kops
+====
+
+The Kubernetes cluster is created and managed using kops_:
+"Production Grade K8s Installation, Upgrades, and Management."
+
+
+Consult the kops documentation to learn more about its operation.
+
+The infrastructure services are defined by the yaml files referenced in the above transcript.
+To ensure repeatability, the services should always be deployed from these version-controlled artifacts.
+The Kubernetes dashboard provides features for directly editing the configuration of the services.
```

Comment:
so... presumably the web interface uses a revision control system for editing configuration?